shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/updatable/ImportDatabaseConfigurationHandler.java

```diff
@@ -94,73 +91,84 @@ protected void update(final ContextManager contextManager) throws DistSQLExcepti
         String databaseName = yamlConfig.getDatabaseName();
         checkDatabaseName(databaseName, file);
         checkDataSource(yamlConfig.getDataSources(), file);
-        updateResources(databaseName, yamlConfig.getDataSources());
-        updateRules(databaseName, yamlConfig.getRules());
+        addDatabase(databaseName);
+        addResources(databaseName, yamlConfig.getDataSources());
+        try {
+            addRules(databaseName, yamlConfig.getRules());
+        } catch (DistSQLException ex) {
+            dropDatabase(databaseName);
+            throw new RuntimeException(ex.getMessage());
```

Comment:
Why is RuntimeException thrown?

Reply:
I made adjustments, please review, thank you.